packages/amplication-data-service-generator/src/server/seed/create-seed.ts

```diff
@@ -67,34 +70,55 @@ export const DEFAULT_AUTH_PROPERTIES = [
   ),
 ];
 
-export async function createSeedModule(userEntity: Entity): Promise<Module> {
-  // eslint-disable-next-line @typescript-eslint/naming-convention
-  const { DTOs, serverDirectories } = DsgContext.getInstance;
-  const MODULE_PATH = `${serverDirectories.scriptsDirectory}/seed.ts`;
-  const file = await readFile(seedTemplatePath);
+export async function createSeed(): Promise<Module[]> {
+  const { serverDirectories, entities } = DsgContext.getInstance;
+
+  const fileDir = serverDirectories.scriptsDirectory;
+  const outputFileName = "seed.ts";
+
+  const userEntity = entities.find((entity) => entity.name === "User")!;
   const customProperties = createUserObjectCustomProperties(userEntity);
 
-  interpolate(file, {
-    DATA: builders.objectExpression([
-      ...DEFAULT_AUTH_PROPERTIES,
-      ...customProperties,
-    ]),
+  const template = await readFile(seedTemplatePath);
+
+  return pluginWrapper(createSeedInternal, EventNames.CreateSeed, {
+    template,
+    seedingProperties: [...DEFAULT_AUTH_PROPERTIES, ...customProperties],
```

Comment:
Why are we using here a different format instead of using "templateMapping" like in all other events?

Reply:
The reason I decided to implement it that way is that I distinguished three things:
1. in this case the template mapping has a specific key: `DATA`
2. the value's type is not any, but `namedTypes.ObjectProperty[]`
3. I think that only with this default and specific implementation of seeding the User entity, we use the `DEFAULT_AUTH_PROPERTIES`. A plugin that makes use of this event should only send the properties that are needed to seed an entity. 
```
const templateMapping = {
    DATA: builders.objectExpression(seedingProperties),
  };
```

Reply:
in all templateMapping we use specific keys - in this case, we only use one key named "DATA".
In case I want to create a new template I should be able to add or update the keys on the object - so, I don't see a reason to use something else

Reply:
"_in all templateMapping we use specific keys - in this case, we only use one key named "DATA"._"
@yuval-hazaz right! 😅 I missed it

Reply:
> "_in all templateMapping we use specific keys - in this case, we only use one key named "DATA"._" @yuval-hazaz right! 😅 I missed it

@abrl91 - please explain how should this information be incorporated in the text